### PR TITLE
Refactored all switch factories to return only an instance of the switch

### PR DESCRIPTION
**Pull request by lindycoder**
SwitchFactory is now in charge of returning a FlowControlSwitch

Aslo added some deprectation in the brocade backward compatibility layer

### CODE REVIEW

**Comment by emmurd** [comment]
LGTM :+1:

File: tests/adapters/switches/dell10g_test.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 from netman.core.objects.switch_transactional import FlowControlSwitch
 
 
+@mock.patch("netman.adapters.switches.dell.warnings.warn", mock.Mock())
```
Comment (fbouliane) [inline tests/adapters/switches/dell10g_test.py]: Could we make deprecations warning inherits from a common ancestor and use a @IgnoreDeprecationWarnings decorator or something like that to ignore deprecations warning ?

Comment (fbouliane) [inline tests/adapters/switches/dell10g_test.py]: Ignore my comment for this PR. Future improvement

Comment (emmurd) [inline tests/adapters/switches/dell10g_test.py]: Good idea :+1:

**Comment by fbouliane** [comment]
LGTM :+1:

**Pull request merged**